src/extension/ScriptRunner.ts

```diff
@@ -187,19 +187,126 @@ export class ScriptRunner {
     })
   }
 
-  private postProblem(str: string): void {
-    const dataList = str.split('\n')
-    const diagnosticDataList = dataList[2].split(':')
-    const rowPosition = parseInt(diagnosticDataList[1]) - 1
-    const colStartPosition = parseInt(diagnosticDataList[2]) - 1
-    const startPosition = new Position(rowPosition, colStartPosition)
-    const colEndPosition = colStartPosition + dataList[3].length - 1
-    const endPosition = new Position(rowPosition, colEndPosition)
-    const range = new Range(startPosition, endPosition)
-    const errorMsg = diagnosticDataList[4]
-    const diagnostic = new Diagnostic(range, errorMsg)
-    const diagnostics: Diagnostic[] = [diagnostic]
-    const uri = Uri.parse(diagnosticDataList[0])
-    this.diagnosticCollection.set(uri, diagnostics)
+  private postProblems(confFile: string): void {
+    const ignoreFolderRegex = '{certora-logs,conf}'
+    const resourceErrorsFile = 'resource_errors.json'
+    vscode.workspace
+      .findFiles(resourceErrorsFile, ignoreFolderRegex, 1)
+      .then(data => {
+        workspace.fs.readFile(data[0]).then(data => {
+          const decoder = new TextDecoder()
+          const content = decoder.decode(data)
+          const resource_error = this.getResourceError(content)
+
+          const pathRegex = /((\\|\/)[a-z0-9\s_@\-^!#$%&+=.{}]+)+/i
+          const locationRegex = /\d+:\d+/g
+
+          resource_error.topics.forEach(topic => {
+            if (topic.messages.length > 0) {
+              topic.messages.forEach(message => {
+                const curMessage: string = message.message
+
+                const path = pathRegex.exec(curMessage)
+                const location = locationRegex.exec(curMessage)
+                const problemPath = this.getPath(path, confFile)
+
+                const descriptiveMessage = (topic.name + ':' + curMessage)
+                  .replace(pathRegex, '')
+                  .replace(locationRegex, '')
+                  .replace(/:{2,}/g, ':')
+
+                const position: Position = this.getPosition(
+                  location,
+                  path,
+                  confFile,
+                )
+                const range: Range = new Range(position, position)
+                const diagnostic = new Diagnostic(range, descriptiveMessage)
+
+                const diagnostics: Diagnostic[] = []
+                diagnostics.push(diagnostic)
+
+                const uri = Uri.parse(problemPath)
+
+                const curDiagnosticCollection: vscode.DiagnosticCollection =
+                  vscode.languages.createDiagnosticCollection()
+
+                curDiagnosticCollection.set(uri, diagnostics)
```

Comment:
Thanks :) some notes about my changes in this part:

- * I had to use async function to get the path to the file where the error originated because of previous changes (checking if a file exists via vscode API) and foreach doest work well with async functions, so I changed it to a "regular" "for loop" and now it works. <-- explanation to why I changed the foreach to for in the new implementation of this part.
- * The rest is very similar to the example you sent.
- * I also changed the part that looks for the resource_errors.json from "than()" to async because the "than" had some annoying behavior.
- * In this part, I also added an error MSG 

> 'Couldn't find `resource_errors.json file. Please contact Certora team'

is that ok?